### PR TITLE
Avoid showing warning messages twice

### PR DESCRIPTION
**Pull request by fotanus**
This is my tentative for fixing #3624.

It don't involve sets :(

### CODE REVIEW

File: lib/bundler/ui/shell.rb [hunk 1]
```diff
@@ -80,6 +81,9 @@ def silence
 
       # valimism
       def tell_me(msg, color = nil, newline = nil)
+        return if @told_messages.include? msg
```
Comment (segiddins) [inline lib/bundler/ui/shell.rb]: this might catch more than warning messages

Comment (fotanus) [inline lib/bundler/ui/shell.rb]: Yeah, the question is whether is desirable to show a message more than once to the user. When intended, the message can be add twice in a row, in the same string. Right?

Comment (segiddins) [inline lib/bundler/ui/shell.rb]: well in theory the same message can be intended twice in different context. Ideally, we'd limit de-duplication just to warning messages

Comment (fotanus) [inline lib/bundler/ui/shell.rb]: I can make that happen if you guys want to merge this way. 

Those are the levels currently:
```
LEVELS = %w(silent error warn confirm info debug)
```
Confirming, should I filter only by warn level?

**Comment by fotanus** [comment]
Please note this avoid showing **any** warning message twice.

File: lib/bundler/ui/shell.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ def initialize(options = {})
         end
         @shell = Thor::Base.shell.new
         @level = ENV['DEBUG'] ? "debug" : "info"
+        @told_messages = []
```
Comment (indirect) [inline lib/bundler/ui/shell.rb]: @fotanus change the name of this ivar to `warning_history` or `past_warnings`, and this looks good to go.

**Comment by fotanus** [comment]
Closing this in favor of #3716 with the requested fix. This is so because I deleted this branch or something :stuck_out_tongue:

**Pull request closed without merging**